File: Chorme_extension/contentScript/All-in-JPEG/content-info/ImageInfo.js
export default class ImageInfo {
    constructor(picture) {
        this.metaDataSize = 0;
        this.offset = 0;
        this.imageDataSize = 0;
        this.attribute = 0;
        this.embeddedDataSize = 0;
        this.embeddedData = [];

        this.init(picture);
    }

    init(picture) {
        
        this.metaDataSize = picture._mataData ? picture._mataData.length : 0;
        this.imageDataSize = picture.imageSize;
        this.attribute = picture.contentAttribute.code;
        this.embeddedDataSize = picture.embeddedSize;

        if (this.embeddedDataSize > 0) {
            this.embeddedData = picture.embeddedData;
        }
    }

    getImageInfoSize() {
        return 20 + this.embeddedDataSize;
    }
}

File: OnePIC/app/src/main/java/com/goldenratio/onepic/AllinJPEGModule/ContentInfo/ImageInfo.kt
package com.goldenratio.camerax.PictureModule.Info

import com.goldenratio.onepic.AllinJPEGModule.Content.Picture

class ImageInfo(picture: Picture) {
    var metaDataSize : Int = 0
    var offset : Int = 0
    var imageDataSize : Int = 0
    var attribute : Int
    var embeddedDataSize : Int = 0
    lateinit var embeddedData : ArrayList<Int>

    init {
        if(picture._mataData != null){
            metaDataSize = picture._mataData!!.size
        }
        imageDataSize = picture.imageSize
        attribute = picture.contentAttribute.code
        embeddedDataSize = picture.embeddedSize
        if(embeddedDataSize > 0)
            embeddedData = picture.embeddedData!!
    }

    fun getImageInfoSize() : Int{
        return 20 + embeddedDataSize
    }
}